reciprocalspaceship/utils/stats.py

```diff
@@ -81,3 +81,41 @@ def compute_redundancy(
 
     mult = mult.sort_index()
     return mult.get_hkls(), mult["Count"].to_numpy(np.int32)
+
+
+def weighted_pearsonr(x, y, w):
+    """
+    Calculate a [weighted Pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient#Weighted_correlation_coefficient).
+
+    Note
+    ----
+    x, y, and w may be arbitrarily batched. the correlation coefficient will be computed pairwise along the last axis.
```

Comment:
I don't quite understand the use of "batched" in this context. Perhaps an example with the corresponding array shapes would help?

Reply:
is this better?